build-system/babel-plugins/babel-plugin-dom-jsx-svg-namespace/index.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview
+ * SVG Elements must be especially created with createElementNS.
+ *
+ * This transforms JSX elements with SVG namespaces so that they include an
+ * '__svg' magic prop in order to create the DOM nodes correctly.
+ *
+ * This prop is consumed by core/dom/jsx. We ignore files that do not import
+ * this module.
+ */
+
+const svgTags = new Set([
+  'animate',
+  'circle',
+  'clipPath',
+  'defs',
+  'desc',
+  'ellipse',
+  'feBlend',
+  'feColorMatrix',
+  'feComponentTransfer',
+  'feComposite',
+  'feConvolveMatrix',
+  'feDiffuseLighting',
+  'feDisplacementMap',
+  'feDistantLight',
+  'feFlood',
+  'feFuncA',
+  'feFuncB',
+  'feFuncG',
+  'feFuncR',
+  'feGaussianBlur',
+  'feImage',
+  'feMerge',
+  'feMergeNode',
+  'feMorphology',
+  'feOffset',
+  'fePointLight',
+  'feSpecularLighting',
+  'feSpotLight',
+  'feTile',
+  'feTurbulence',
+  'filter',
+  'foreignObject',
+  'g',
+  'image',
+  'line',
+  'linearGradient',
+  'marker',
+  'mask',
+  'metadata',
+  'path',
+  'pattern',
+  'polygon',
+  'polyline',
+  'radialGradient',
+  'rect',
+  'stop',
+  'svg',
+  'switch',
+  'symbol',
+  'text',
+  'textPath',
+  'tspan',
+  'use',
+  'view',
+]);
+
+module.exports = function (babel) {
+  const {types: t} = babel;
+
+  let isCoreDomJsx = false;
+  return {
+    name: 'dom-jsx-svg-namespace',
+    visitor: {
+      Program: {
+        enter(path) {
+          isCoreDomJsx = false;
+          path.traverse({
+            ImportDeclaration(path) {
+              if (path.node.source.value.endsWith('/core/dom/jsx')) {
+                isCoreDomJsx = true;
+              }
+            },
+          });
+        },
+      },
+      JSXOpeningElement(path) {
+        if (!isCoreDomJsx) {
+          return;
+        }
+        const {name} = path.node.name;
+        if (svgTags.has(name)) {
+          path.node.attributes = path.node.attributes.filter(
+            // We can remove this for bundle size since the renderer will
+            // create the element with a given namespace anyway.
+            (attr) => attr.name.name !== 'xmlns'
+          );
```

Comment:
@jridgewell Would this be safe to do? Seems to be rendering fine in Chrome if we remove the attribute.

Reply:
Safe to remove, though this should just be a lint step?

Reply:
I disagree about linting, a generally correct source requires the attribute.

Reply:
Not removing `xmlns` anymore, but setting it every time. See https://github.com/ampproject/amphtml/pull/36722#discussion_r742169459